daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -399,6 +400,11 @@ class Runner(compiledPackages: CompiledPackages, script: Script.Action, timeMode
                           } yield v
                         }
                         case Left(statusEx) => {
+                          // This branch is superseded by SubmitMustFail below,
+                          // however, it is maintained for backwards
+                          // compatibility with DAML script DARs generated by
+                          // older SDK versions that didn't distinguish Submit
+                          // and SubmitMustFail.
```

Comment:
I have confirmed that replacing this branch by `Future.failed(...)` breaks the corresponding compatibility test.

Reply:
We don’t seem to have a `submitMustFail` in the compat tests. Should we add one?

Reply:
Oh nvm, I just saw that you added it :facepalm: